cores/arduino/stm32/uart.c

```diff
@@ -128,10 +128,14 @@ void uart_init(serial_t *obj, uint32_t baudrate, uint32_t databits, uint32_t par
 
   /* Determine the U(S)ART peripheral to use (USART1, USART2, ...) */
   USART_TypeDef *uart_tx = pinmap_peripheral(obj->pin_tx, PinMap_UART_TX);
-  USART_TypeDef *uart_rx = pinmap_peripheral(obj->pin_rx, PinMap_UART_RX);
+  USART_TypeDef *uart_rx = NP;
```

Comment:
If `obj->pin_rx` is `NC ` then `uart_rx` will be `NP`.
So I fdon't think this change is required.